pkg/machine/vmconfigs/machine.go

```diff
@@ -185,28 +185,32 @@ func (mc *MachineConfig) Remove(saveIgnition, saveImage bool) ([]string, func()
 	}
 
 	mcRemove := func() error {
+		if err := connection.RemoveConnections(mc.Name, mc.Name+"-root"); err != nil {
+			return err
```

Comment:
if we return right away here it will not delete the other things on error which is bad

This here should really use some of the multierror packages that we use and just store all errors and then as last step remove it, i.e. declare a `errs []error` slice at the top then append all error there instead of returning. Then at the function end return `errorhandling.JoinErrors(errs)`.

Reply:
I didn't know that existed, thanks Paul. Should be fixed as well.